dune_client/api/extensions.py

```diff
@@ -0,0 +1,196 @@
+"""
+Extended functionality for the ExecutionAPI
+"""
+
+import time
+from typing import Union, Optional, Any
+
+from deprecated import deprecated
+
+from dune_client.api.execution import ExecutionAPI
+from dune_client.models import (
+    ResultsResponse,
+    DuneError,
+    ExecutionState,
+    QueryFailed,
+    ExecutionResultCSV,
+)
+from dune_client.query import QueryBase
+
+
+class ExtendedAPI(ExecutionAPI):
+    """
+    Provides higher level helper methods for faster
+    and easier development on top of the base ExecutionAPI.
+    """
+
+    def run_query(
+        self,
+        query: QueryBase,
+        ping_frequency: int = 5,
```

Comment:
As true as this may be, this would be a breaking change to the existing api and I am not sure its really worth it. 
It is unlikely that many people are actually setting this essentially optional field, but I do know of at least one project that is.

https://github.com/cowprotocol/solver-rewards/blob/700fa71e9c79bd02784b6e9eee0de1b8e96da4d6/src/fetch/dune.py#L57

Reply:
I think it's really a nitpick and it doesn't absolutely have to be changed, considering this would indeed be a breaking change.

As to whether people know this is seconds, I think it depends on the context and expectations. For instance, if someone is not very familiar with Dune and expect queries to run in milliseconds, it might be surprising to them that these are in fact seconds.

Reply:
I think I'll skip this one for now. I could add an comment or maybe rework the doc-strings to make it more clear at some point. The slightly unfortunate thing is that this parameter appears in multiple places...